x/mongo/driver/topology/fsm.go

```diff
@@ -106,7 +89,37 @@ func (f *fsm) apply(s description.Server) (description.Topology, description.Ser
 		updatedDesc = f.applyToSingle(s)
 	}
 
-	return f.Topology, updatedDesc, nil
+	for _, server := range f.Servers {
+		if server.WireVersion != nil {
+			if server.WireVersion.Max < supportedWireVersions.Min {
+				f.compatible.Store(false)
+				f.compatibilityErr = fmt.Errorf(
+					"server at %s reports wire version %d, but this version of the Go driver requires "+
+						"at least %d (MongoDB %s)",
+					server.Addr.String(),
+					server.WireVersion.Max,
+					supportedWireVersions.Min,
+					minSupportedMongoDBVersion,
+				)
+				return description.Topology{}, s
+			}
+
+			if server.WireVersion.Min > supportedWireVersions.Max {
+				f.compatible.Store(false)
+				f.compatibilityErr = fmt.Errorf(
+					"server at %s reports wire version %d, but this version of the Go driver requires "+
```

Comment:
This error message isn't correct. Per https://github.com/mongodb/specifications/blob/master/source/server-discovery-and-monitoring/server-discovery-and-monitoring.rst#checking-wire-protocol-compatibility, it should be `server at <address> requires wire version <server.WireVersion.Min>, but this version of the Go driver only supports up to <supportedWireVersions.Max>`.